src/lib/blkalloc/append_blk_allocator.cpp

```diff
@@ -190,7 +150,13 @@ blk_num_t AppendBlkAllocator::get_used_blks() const { return m_last_append_offse
 
 blk_num_t AppendBlkAllocator::get_freeable_nblks() const { return m_freeable_nblks; }
 
-blk_num_t AppendBlkAllocator::get_defrag_nblks() const { return get_freeable_nblks() - available_blks(); }
+blk_num_t AppendBlkAllocator::get_defrag_nblks() const { return get_freeable_nblks() + available_blks(); }
```

Comment:
if I understand correctly, `get_defrag_nblks()` will return how many blks can be GC, so in your PR,  should just return `m_freeable_nblks`, right?

Reply:
I am not sure why we need get_freeable_blks() and get_defrag_nblks() return the same thing. I would expect get_defrag_blks() indicate how much I get after I defragment them, right?

Reply:
No, we shouldn't make this change. defrag_nblks should exclude available blocks, otherwise, if it is a chunk with nothing written on it yet, it will also be selected for GC, which is wrong. 

Defrag_nblks should only be the number of blocks, that by compacting the shard to a new place, that how many nblks could be freed, by nature it shouldn't include available blocks on this chunk. This is the reason why m_freeable_nblks is init to all available blks so that it can keep track of the defrag nblks.

Reply:
Agreed and updated with latest changes, please rereview them.